ios/Classes/SwiftWorkmanagerPlugin.swift

```diff
@@ -1,14 +1,135 @@
 import Flutter
 import UIKit
+import os
 
-public class SwiftWorkmanagerPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "workmanager", binaryMessenger: registrar.messenger())
-    let instance = SwiftWorkmanagerPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
+public class SwiftWorkmanagerPlugin: FlutterPluginAppLifeCycleDelegate {
+    
+    private struct ForegroundMethodChannel {
+        static let channelName = "be.tramckrijte.workmanager/foreground_channel_work_manager"
+        enum methods: String {
+            case initialize
+            case performSampleFetch
```

Comment:
Or we could use the OptionalMethodChannel in Dart.

```
/// A [MethodChannel] that ignores missing platform plugins.
///
/// When [invokeMethod] fails to find the platform plugin, it returns null
/// instead of throwing an exception.
```